### ISSUE TITLE
During training, rtdetr cannot be made to set amp to false.

### ISSUE THREAD

**Issue by tonycc521** [ultralytics/ultralytics#21105]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

During training, rtdetr cannot be made to set amp to false.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tonycc521, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will assist you here soon.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for comprehensive guidance, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions about training and model options are answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug the issue with RT-DETR and the amp setting.

If your question is about custom training, please include as much detail as possible (such as dataset images, config files, and training logs) and check our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our community for support and discussion:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for deeper discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to connect with other users

## Upgrade

To ensure you’re using the latest features and bug fixes, upgrade the `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO runs smoothly in these verified environments (with all dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM ([GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/))
- **Amazon** Deep Learning AMI ([AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/))
- **Docker Image** ([Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience!

**Comment by Y-T-G** [comment]
I am not able to reproduce this with latest Ultralytics

**Comment by tonycc521** [comment]
results = model.train(
                data=dataset_yaml,
                imgsz=self.img_size,
                epochs=self.epochs,
                batch=self.batch_size,
                workers=8,  
                device=0,
                optimizer=self.optimizer,
                lr0=self.learning_rate,
                lrf=self.lr_decay,
                momentum=self.momentum,
                weight_decay=self.weight_decay,
                patience=self.patience,  
                close_mosaic=10,
                resume=False,
                project=self.save_dir,
                name=self.model_name,
                exist_ok=True,
                single_cls=False,
                cache=True, 
                amp=True,  
                cos_lr=True,
                augment=True,
                save_period=10,
                hsv_h=getattr(self, 'hsv_h', 0.015),
                hsv_s=getattr(self, 'hsv_s', 0.7),
                hsv_v=getattr(self, 'hsv_v', 0.4),
                degrees=getattr(self, 'degrees', 0.0),
                translate=getattr(self, 'translate', 0.1),
                scale=getattr(self, 'scale', 0.5),
                shear=getattr(self, 'shear', 0.0),
                fliplr=getattr(self, 'flip_left_right', 0.5),
                flipud=getattr(self, 'flip_up_down', 0.0),
                verbose=False,
            )
This is the way I set it up.

**Comment by Y-T-G** [comment]
What happens if you set `amp=False`?
